### PR TITLE
usePopState -> useLocationChange

### LINKED ISSUE #28: usePopState has wrong type definition, hook dependencies
Hey again!

So I have a use-case where a modal dialog contains links and the modal needs to close on any history change.

Rather than making my own listener, I went to see how `raviger` implements it, and found the nifty `usePopState` hook that does pretty much what I need. However I found the following:

1. `usePopState` is exposed but has no entry in the documentation. Is it part of the API? Is it fine to use it?

2. In the type definition, [it says](https://github.com/kyeotic/raviger/blob/master/types/raviger.d.ts#L44) that the `predicate` parameter is a function that returns a boolean, however, according to [the actual implementation](https://github.com/kyeotic/raviger/blob/master/src/path.js#L24) and usages across the library, it seems that it's only expecting a boolean.

3. The `predicate` parameter is not added to the dependencies of the `useEffect` hook, which means that if we need to change that condition after the initial render, it won't be taken into account.

4. I feel that `predicate` should be an optional argument, and could be moved to being the third parameter?

5. Probably have the same feeling about the first parameter. Seems like `basePath` could be optional and default to `''`.

Looking forward to hearing about your opinion on these 👍

### PR DESCRIPTION
**Pull request by kyeotic**
rename `usePopState` to `useLocationChange` and rework the parameters.

Fixes #28

### CODE REVIEW

File: types/raviger.d.ts [hunk 1]
```diff
@@ -39,10 +39,13 @@ export function usePath(basePath?: string): string
...
-  basePath: string,
-  predicate: () => boolean,
-  setFn: (path: string) => any
+export function useLocationChange(
+  setFn: (path: string) => any,
+  options?: {
+    inheritBasePath: boolean
+    basePath: string
+    isActive: () => boolean | boolean
+  }
```
Comment (yenoiwesa) [inline types/raviger.d.ts]: Shouldn't each `options` properties be marked as optional as well?

File: src/path.js [hunk 2]
```diff
@@ -19,15 +23,25 @@ export function getCurrentPath(basePath = '') {
     : window.location.pathname.replace(basePathMatcher(basePath), '') || '/'
 }
 
-export function usePopState(basePath, predicate, setFn) {
+export function useLocationChange(setFn, options = {}) {
+  if (isNode) return
+  let basePath = ''
+  if (options.inheritBasePath !== false) basePath = useBasePath()
```
[the remark this answers is not in the document]
Comment (kyeotic) [reply src/path.js]: Damn, good catch. Wonder why my linter didn't yell at me

File: src/router.js [hunk 3]
```diff
@@ -17,9 +17,14 @@ export function useRoutes(
   const [context, setContext] = useState({ path })
 
   // Watch for location changes
-  usePopState(basePath, isNode, path =>
-    setContext(context => ({ ...context, path }))
+  let pathWatcher = useCallback(
```
[the remark this answers is not in the document]
Comment (kyeotic) [reply src/router.js]: You've noticed I still favor `let`... bad habit :)

**Comment by kyeotic** [comment]
@yenoiwesa Thanks for the review! I've updated per your comments

**Comment by kyeotic** [comment]
@yenoiwesa I'll merge and republish when you ✅, or tonight, whichever comes first

**Comment by kyeotic** [comment]
published under `1.0.0-rc.2`

**Pull request merged**